### PR TITLE
add update and delete to the reaction

### PR DESCRIPTION
**Pull request by RufiMohammed**
add update and delete to the reaction

### CODE REVIEW

**Comment by RufiMohammed** [comment]
add update and delete to the reaction

File: app/Http/Controllers/Api/ReactionController.php [hunk 1]
```diff
@@ -30,109 +30,190 @@ public function index()
...
+        ####Rufi####
+         //validate requested data
         $validator = Validator::make($request->all(), [
-            'reaction_id'   => 'required',
-            'comment_id' => 'required_without:post_id',
-            'post_id'    => 'required_without:comment_id',
+            'reaction_id'  => 'required',
+            'comment_id'   => 'required_without_all:post_id,media',
+            'post_id'      => 'required_without_all:comment_id,media',
+            'media'        => 'required_without_all:post_id,comment_id',
```
Comment (saraIsmailSE) [inline app/Http/Controllers/Api/ReactionController.php:116]: Media is required where reaction id is zero

File: app/Http/Controllers/Api/ReactionController.php [hunk 2]
```diff
@@ -30,109 +30,190 @@ public function index()
...
         }
-        if($request->has('comment_id'))
-         $reaction = Reaction::where('user_id', Auth::id())->where('comment_id', $request->comment_id)->first();
-        else if($request->has('post_id'))
-         $reaction = Reaction::where('user_id', Auth::id())->where('post_id', $request->post_id)->first();
-        if($reaction){
-            $reaction->update($request->all());
-            return $this->jsonResponseWithoutMessage("Reaction Updated Successfully", 'data', 200);
+        //******To edit media of reaction by user have permission to edit******//
+        if($request->has('media')){
```
Comment (saraIsmailSE) [inline app/Http/Controllers/Api/ReactionController.php:123]: If reaction id is zero
You can do find reaction first then use reaction->reaction_id to check permission

**Review by saraIsmailSE — COMMENTED** [review]
Done

**Pull request merged**